### PR TITLE
37 housing fund

### PR DESCRIPTION
**Pull request by letodunc**

### CODE REVIEW

**Review by ndkazu — APPROVED** [review]

File: pallets/housing_fund/src/structs.rs [hunk 1]
```diff
@@ -25,6 +25,49 @@ pub enum WithdrawalReason {
...
+        // add to transferable
+        self.transferable += amount.clone();
+        // update the total amount
+        self.total += amount.clone();
+    }
+
+    pub fn can_withdraw(&self, amount: BalanceOf<T>) -> bool {
+        // check that amount to withdraw if inferior to the transferable
+        amount.clone() <= self.transferable.clone()
+    }
```
Comment (ilhanu) [inline pallets/housing_fund/src/structs.rs:53]: So this looks like in the right direction, we can discuss in a team meeting. So there are currency traits available in substrate  such as [Lockable](https://paritytech.github.io/substrate/master/frame_support/traits/trait.LockableCurrency.html) or [Reservable](https://paritytech.github.io/substrate/master/frame_support/traits/trait.ReservableCurrency.html)

The fund should reserve & lock the currency, but that also means the users that contributed contributed and selected after the fair algo selection ( which the pallet is not available atm) should ignite this.

File: pallets/housing_fund/src/lib.rs [hunk 2]
```diff
@@ -181,7 +185,26 @@ pub mod pallet {
...
+			ensure!(amount.clone() <= contribution_amount, Error::<T>::NotEnoughFundToWithdraw);
+
+			// Get the block number for timestamp
+			let block_number = <frame_system::Pallet<T>>::block_number();
+
+			Contributions::<T>::mutate(&who, |val| {
+				let unwrap_val = val.clone().unwrap();
+				let contribution_logs = unwrap_val.contributions.clone();
+
+				let contrib = Contribution {
```
Comment (ilhanu) [inline pallets/housing_fund/src/lib.rs:237]: please use full words for variables

File: pallets/housing_fund/Cargo.toml [hunk 3]
```diff
@@ -3,11 +3,11 @@ name = "pallet-housing_fund"
 version = "4.0.0-dev"
 description = "FRAME pallet template for defining custom runtime logic."
 authors = ["Substrate DevHub <https://github.com/substrate-developer-hub>"]
-homepage = "https://fairsquares.io/"
```
Comment (ilhanu) [inline pallets/housing_fund/Cargo.toml]: it's https://fair-squares.nl

**Review by ilhanu — CHANGES_REQUESTED** [review]
Great start for the housing fund pallet! It would be nice if a PR is openend it either has a description or references to the issue   #37 . It would also be nice if the issue checkboxes are completed then we can look for what we have finished and what we are reviewing. 

Please see my comment about the Reservable and Lockable Currency as these traits are already wired to our Currency trait and it's works in the box but we have to wire them. 

Please also unit-test, it would really help to kick-start the testing suite. For the rest awesome work 🥇 💯 👍🏼 🚀

**Pull request merged**